### ISSUE TITLE
Support for Helm charts (templated YAML)

### ISSUE THREAD

**Issue by jackwhelpton** [sbaudoin/sonar-yaml#46]
A common use of YAML within an application is when defining a Helm chart. These bear a YAML extension, but typically result in two violations. Firstly, I tend to set up the linter to enforce a single space after opening braces and before closing ones. Because `{{` is used as a templating expression, code such as this:

```
labels:
  app: {{ template "my-application.name" . }}
```

results in "too few spaces inside braces (braces)". I could allow 0 _or_ 1 space, which I may be forced to do, but it'd be great if there was a way to override this check if the next character is also a brace, allowing `{{`.

In a related fashion, this line is flagged with "Parse error: syntax error: expected the node content, but found '-'":

```
{{- $context := index .Values "current-context" | default "dev" }}
```

`{{-` is used in Helm to trim whitespace. Is there a way to tweak the rules to avoid flagging this construct as an error?

**Comment by jackwhelpton** [comment]
Having cloned the repo just to get my head around the architecture, it seems like this should be raised as an issue for the `yamllint` library... I'll raise it there too, this can be left to track any integration changes that are required, or it can be closed.

**Comment by sbaudoin** [comment]
Hello,

Thanks, yes this is a yamllint issue. The problem is that the YAML linter is really "dumb": it checks YAML literally and it knows nothing about templating. Not sure I can fix anything there...

**Comment by sbaudoin** [comment]
See sbaudoin/yamllint#16: the solution so far, as it is not possible to analyze with yamllint or this plugin the YAML produced by the template engines, consists in excluding the templates from the analysis. For this, you have to have a `.yamllint` configuration file in your project that will have `ignore` parameters to exclude the templates.
